shepherd_bms/include/datastructs.h

```diff
@@ -1,5 +1,66 @@
 #ifndef DATASTRUCTS_H
 #define DATASTRUCTS_H
 
+#include <nerduino.h>
+
+
+
+/*Individual chip data*/
+typedef struct ChipData_t
+{
+    uint16_t VoltageReading[12];          //store voltage readings from each chip
+    uint16_t thermistorReading[32];       //store all therm readings from each chip
+    bool discharge[12];
+};
+
+typedef enum BMSFault_t
+{
+    //Orion BMS faults
+    CELLS_NOT_BALANCING;
+    CELL_VOLTAGE_TOO_HIGH;
+    CELL_VOLTAGE_TOO_LOW;
+    CELL_ASIC_FAULT;                    //internal fault with the cell voltage measurement circuitry
+    PACK_TOO_HOT;
+    CHARGE_INTERLOCK_FAULT;             //both Charge Power (pin 3) and Ready Power (pin 2) are energized at the same time
+    OPEN_WIRING_FAULT;                  //cell tap wire is either weakly connected or not connected 
+    PACK_VOLTAGE_MISMATCH;              //total pack voltage sensor did not match the sum of the individual cell voltage measurements
+    INTERNAL_SOFTWARE_FAULT;
+    INTERNAL_HEATSINK_THERMISTOR;       //internal hardware fault has occurredwith the internal thermistors that monitor the unit temperature
```

Comment:
What will this be for us?

Reply:
We can measure the temp/humidity with the chip on NERduino, might rename tho

Reply:
Ah yes. That checks out